pyccel/codegen/printing/ccode.py

```diff
@@ -317,9 +319,10 @@ def _print_PyccelMul(self, expr):
         return ' * '.join(self._print(a) for a in expr.args)
 
     def _print_PyccelDiv(self, expr):
-        args = [self._print(a) for a in expr.args]
+        args = expr.args
         if all(a.dtype is NativeInteger() for a in expr.args):
-            return ' / '.join('real({})'.format(self._print(a)) for a in args)
+            args = [PythonFloat(a) for a in args]
+        args = [self._print(a) for a in args]
         return  ' / '.join(self._print(a) for a in args)
```

Comment:
In this function `args` is redefined multiple times, but there is no reason to do so. Moreover, the printer processes the elements of `args` twice. Instead, you could write it as follows:
```python
def _print_PyccelDiv(self, expr):

    if all(a.dtype is NativeInteger() for a in expr.args):
        args = [PythonFloat(a) for a in expr.args]
    else:
        args = expr.args

    return  ' / '.join(self._print(a) for a in args)
```